src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/TypeLoaderExports.cs

```diff
@@ -162,8 +163,20 @@ private static unsafe Value CacheMiss(IntPtr context, IntPtr signature, RuntimeO
     internal static unsafe class RawCalliHelper
     {
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void Call(System.IntPtr pfn, ref byte data)
-            => ((delegate*<ref byte, void>)pfn)(ref data);
+        public static void CallDefaultStructConstructor(System.IntPtr pfn, ref byte data)
+        {
+            // Manually expand call of the instance method fat pointer. We cannot use a regular C# function
+            // pointer call because it cannot express call of an instance method.
```

Comment:
Is it "instance method" or "generic method"? If the former, how is is the non generic method pointer case working?

Reply:
The non-generic constructor has a single argument (this pointer). This argument is always passed in the same register, and so the instance vs. static function pointer can be used interchangeably.

The shared generic constructor has two arguments - this pointer and hidden generic dictionary arg in this order. The static shared generic method that takes a single byref argument has two arguments too, but the hidden generic dictionary arg is first. The difference in the ordering of these two arguments means that it is not possible use the instance vs. static function pointer interchangeably.

Reply:
Okay so this seems to be about both instance vs static WITH generic methods. It isn't exclusively related to instance methods. The generic rules are a key part of this.

Reply:
I have updated the comment to mention shared generics.